src/app/modules/submission/email-submission/ParsedResponsesObject.class.ts

```diff
@@ -23,11 +28,38 @@ export default class ParsedResponsesObject {
   public ndiResponses: ProcessedFieldResponse[] = []
   private constructor(public responses: ProcessedFieldResponse[]) {}
 
-  addNdiResponses(
-    ndiResponses: ProcessedFieldResponse[],
-  ): ParsedResponsesObject {
-    this.ndiResponses = ndiResponses
-    return this
+  addNdiResponses({
+    authType,
+    uinFin,
+    userInfo,
+  }: {
+    authType: AuthType
+    uinFin: string
+    userInfo?: string
```

Comment:
might it be better to type these more strictly? perhaps something like
```
type NdiResponseInfo = 
  | { authType: AuthType.SP | AuthType.MyInfo; uinFin: string }
  | { authType: AuthType.CP; uinFin: string; userInfo: string }
```
so we don't end up e.g. accidentally adding a Corppass response without a `userInfo`

Reply:
oh wow didnt think of that! I've just incorporated this change, now I don't have to use neverthrow since the types guarantee the correct permutation is always passed in.